src/components/ModalInvites.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { connect, useDispatch } from 'react-redux'
+import { InviteCodes, State } from '../@types'
+import InvitesIcon from './icons/InvitesIcon'
+import CheckmarkIcon from './icons/CheckmarkIcon'
+import CopyClipboard from './icons/CopyClipboard'
+import { alert, createUserInvites } from '../action-creators'
+import { isTouch } from '../browser'
+import { theme } from '../selectors'
+import { createId, timestamp } from '../util'
+import { ActionButton } from './ActionButton'
+import Modal from './Modal'
+import Input from './Input'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const mapStateToProps = (state: State) => {
+  const { authenticated, user: { uid = '' } = {}, userInvites } = state
+  return {
+    dark: theme(state) !== 'Light',
+    uid,
+    authenticated,
+    userInvites,
+  }
+}
+
+/** Modal to get gift codes. */
+const ModalInvites = ({ dark, uid, authenticated, userInvites }: ReturnType<typeof mapStateToProps>) => {
+  let giftCodes: { link: string; used: boolean }[] = []
+  const isMounted = useRef(false)
+  const dispatch = useDispatch()
+
+  const [inputType, updateInputType] = useState(['password', 'password', 'password'])
+
+  const [selectedGiftCode, updateSelectedGiftCode] = useState(-1)
+  const [copiedGiftCode, updateCopiedGiftCode] = useState('')
+  const [giftCode, updateGiftCode] = useState(giftCodes)
+
+  let inviteCodeCnt = 0
+
+  // eslint-disable-next-line jsdoc/require-jsdoc
+  const generateInviteCode = () => {
+    if (inviteCodeCnt > 2) return
+
+    inviteCodeCnt++
+
+    const inviteId = createId().slice(0, 8)
+    const newInviteCode: InviteCodes = {
+      createdBy: uid,
+      created: timestamp(),
+    }
+
+    dispatch(createUserInvites(newInviteCode, inviteId, uid))
+
+    generateInviteCode()
+  }
+
+  // eslint-disable-next-line jsdoc/require-jsdoc
+  const createGiftCode = (invite: InviteCodes = {}) => {
+    giftCodes = [
+      ...giftCodes,
+      {
+        link: `${window.location.origin}/signup?code=${invite.id}`,
+        used: !!invite.used,
+      },
+    ]
+
+    updateGiftCode(giftCodes)
+  }
+
+  useEffect(() => {
+    if (authenticated) {
+      if (!isMounted.current) {
+        isMounted.current = true
+        if (userInvites && userInvites.length === 0) {
+          generateInviteCode()
+        } else {
+          userInvites &&
+            userInvites.forEach((invite: InviteCodes) => {
+              return createGiftCode(invite)
+            })
```

Comment:
Don't use `forEach`. Use `Array.map` to generate an array of `giftCodes` and then update the state using `updateGiftCodes`at once. Iterating and calling `createGiftCode` is not necessary.